### PR TITLE
Switch from deprecated pkg `k8s.io/utils/pointer` to recommended `k8s.io/utils/ptr` for pointer operations

### LINKED ISSUE #863: Use pointer utility pkg `k8s.io/utils/ptr` and remove the deprecated pkg `k8s.io/utils/pointer`
**How to categorize this issue?**

/area quality
/kind enhancement

**What would you like to be added**:

The pkg "k8s.io/utils/pointer" is deprecated starting from k8s.io/utils version `v0.0.0-20240502163921-fe8a2dddb1d0` which is our current version in master. It is recommended to use the alternate pkg "k8s.io/utils/ptr" from here on. 

Remove the deprecated package and use the recommended package for pointer related operations.

Check the release timeline for `k8s.io/utils` [here](https://pkg.go.dev/k8s.io/utils@v0.0.0-20240502163921-fe8a2dddb1d0?tab=versions)

**Why is this needed**:
The support for "k8s.io/utils/pointer" pkg will be removed going further as it is currently in deprecated state. So it's best to use the newly introduced pkg "k8s.io/utils/ptr" for pointer operations.

### PR DESCRIPTION
**Pull request by anveshreddy18**
**How to categorize this PR?**

/area quality
/kind enhancement

**What this PR does / why we need it**:

The pkg "k8s.io/utils/pointer" is deprecated starting from k8s.io/utils version `v0.0.0-20240502163921-fe8a2dddb1d0` which is our current version in master. It is recommended to use the alternate pkg "k8s.io/utils/ptr" from here on. 

This PR removes the deprecated package and uses the recommended package for pointer related operations.

Check the release timeline for `k8s.io/utils` [here](https://pkg.go.dev/k8s.io/utils@v0.0.0-20240502163921-fe8a2dddb1d0?tab=versions)

**Which issue(s) this PR fixes**:
Fixes #863

**Special notes for your reviewer**:

**Release note**:

```other developer
remove deprecated pkg `k8s.io/utils/pointer` and use pkg `k8s.io/utils/ptr`
```

### CODE REVIEW

**Review by renormalize — APPROVED** [review]
thanks!

**Comment by unmarshall** [comment]
@anveshreddy18 since we did not have a generic function to get a pointer or dereference a pointer, that is why we introduced a function [PointerOf](https://github.com/gardener/etcd-druid/blob/master/internal/utils/miscellaneous.go#L76-L78) and [TypeDeref](https://github.com/gardener/etcd-druid/blob/master/internal/utils/miscellaneous.go#L55-L59). Please check if these can now be removed.

File: api/v1alpha1/helper.go [hunk 1]
```diff
@@ -106,10 +106,10 @@ func GetNamespaceName(etcdObjMeta metav1.ObjectMeta) types.NamespacedName {
 // to suspend spec reconciliation for this Etcd resource. If no annotation is set then it will return nil.
 func GetSuspendEtcdSpecReconcileAnnotationKey(etcdObjMeta metav1.ObjectMeta) *string {
 	if metav1.HasAnnotation(etcdObjMeta, SuspendEtcdSpecReconcileAnnotation) {
-		return pointer.String(SuspendEtcdSpecReconcileAnnotation)
+		return ptr.To(SuspendEtcdSpecReconcileAnnotation)
```
Comment (unmarshall) [inline api/v1alpha1/helper.go:109]: Please look at all usages of `PointerOf` and `TypeDeref` and now you can replace these with `ptr.To` and `ptr.Deref` functions.

**Review by unmarshall — CHANGES_REQUESTED** [review]

File: api/v1alpha1/helper.go (hunk 1 above)
Comment (anveshreddy18) [reply api/v1alpha1/helper.go:109]: Thanks for pointing this to me. I was not aware of this. 

Now I've addressed this concern. PTAL

**Review by unmarshall — APPROVED** [review]

**Pull request merged**